src/coins.cpp

```diff
@@ -45,27 +44,25 @@ size_t CCoinsViewCache::DynamicMemoryUsage() const {
 CCoinsMap::iterator CCoinsViewCache::FetchCoin(const COutPoint &outpoint) const {
     const auto [ret, inserted] = cacheCoins.try_emplace(outpoint);
     if (inserted) {
-        if (!base->GetCoin(outpoint, ret->second.coin)) {
+        if (auto coin{base->GetCoin(outpoint)}) {
+            if (coin->IsSpent()) { // TODO GetCoin cannot return spent coins
+                // The parent only has an empty entry for this outpoint; we can consider our version as fresh.
+                ret->second.AddFlags(CCoinsCacheEntry::FRESH, *ret, m_sentinel);
+            }
+            cachedCoinsUsage += coin->DynamicMemoryUsage();
+            ret->second.coin = std::move(*coin);
```

Comment:
This line should be moved up to just under the if statement to keep behavior the same. The coin is already moved into `ret->second.coin` before we potentially call `AddFlags`.

Although this will hopefully go away in #30673.

Reply:
I don't think there's any difference (except for having to use `ret->second.coin.DynamicMemoryUsage` and `ret->second.coin.IsSpent()` instead), but I don't mind moving it up.